rootex/framework/systems/render_system.cpp

```diff
@@ -132,6 +132,20 @@ void RenderSystem::renderPassRender(float deltaMilliseconds, RenderPass renderPa
 		}
 	}
 
+	for (auto& mc : ECSFactory::GetAllSpriteComponent())
+	{
+		if (mc.getRenderPass() & (unsigned int)renderPass)
+		{
+			mc.preRender(deltaMilliseconds);
+			if (mc.isVisible())
+			{
+				Vector3 viewDistance = mc.getTransformComponent()->getAbsolutePosition() - m_Camera->getAbsolutePosition();
+				mc.render(viewDistance.Length());
+			}
+			mc.postRender();
+		}
+	}
+
```

Comment:
won't be needed

Reply:
Given that we need this for GridModelComponent which inherits from ModeComponent, I suppose we will be needing this for SpriteComponent as well?

Reply:
right, leave it be